nodedb.go

```diff
@@ -237,6 +237,53 @@ func (ndb *nodeDB) DeleteVersionsFrom(version int64) error {
 	return nil
 }
 
+// DeleteVersionsTo deletes all versions below the given height.
+func (ndb *nodeDB) DeleteVersionsTo(version int64) error {
+	if version == 0 {
+		return errors.New("version must be greater than 0")
+	}
+
+	latest := ndb.getLatestVersion()
+	if latest < version {
+		return errors.Errorf("cannot delete latest saved version (%d)", version)
+	}
+
+	root, err := ndb.getRoot(version)
+	if err != nil {
+		return err
+	}
+	if root == nil {
+		return errors.Errorf("root for version %v not found", version)
+	}
+
+	for v, r := range ndb.versionReaders {
+		if v < version && r != 0 {
+			return errors.Errorf("unable to delete version %v with %v active readers", v, r)
+		}
+	}
+
+	ndb.mtx.Lock()
+	defer ndb.mtx.Unlock()
+
+	// Delete orphans
+	ndb.traverseOrphans(func(key, hash []byte) {
+		var fromVersion, toVersion int64
+		orphanKeyFormat.Scan(key, &toVersion, &fromVersion)
+		if toVersion < version {
```

Comment:
```suggestion
		if fromVersion < version {
```
Is it possible to do this?
It seems to me that this is not logically correct, but the tests pass. Apparently, this is a bottleneck for me, could you explain to me?